devicetypes/smartthings/philio-multiple-sound-siren.src/philio-multiple-sound-siren.groovy

```diff
@@ -0,0 +1,305 @@
+ * Range: 0-127
+ */
+def getDurationMap() {[
+	"Never stop": 0,
+	"30 seconds": 1,
+	"1 minute": 2,
+	"2 minutes": 4,
+	"3 minutes": 6,
+	"5 minutes": 10,
+	"10 minutes": 20,
+	"20 minutes": 40,
+	"30 minutes": 60,
+	"45 minutes": 90,
+	"1 hour": 120
+]}
+
+def getDefaultSound() { "Emergency" }
+def getDefaultDuration() { "3 minutes" }
+
+def installed() {
+	// Device-Watch simply pings if no device events received for 32min(checkInterval)
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+
+	state.sound = defaultSound
+	state.duration = defaultDuration
+
+	// Get default values
+	response(secure(zwave.basicV1.basicGet()))
+}
+
+def updated() {
+	def commands = []
+	// Device-Watch simply pings if no device events received for 32min(checkInterval)
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+
+	log.debug "settings: ${settings.inspect()}, state: ${state.inspect()}"
+
+	def sound = settings.sound ?: state.sound
+	def duration = settings.duration ?: state.duration
+
+	if (sound != state.sound || duration != state.duration) {
+		state.sound = sound
+		state.duration = duration
+		commands << secure(zwave.configurationV1.configurationSet(parameterNumber: 31, size: 1, configurationValue: [durationMap[duration]]))
+	}
+
+	response(commands)
+}
+
+def parse(String description) {
+	log.debug "parse($description)"
+	def result = null
+	if (description.startsWith("Err")) {
+		if (zwInfo?.zw?.contains("s")) {
+			result = createEvent(descriptionText:description, displayed:false)
+		} else {
+			result = createEvent(
+					descriptionText: "This device failed to complete the network security key exchange. If you are unable to control it via SmartThings, you must remove it from your network and add it again.",
+					eventType: "ALERT",
+					name: "secureInclusion",
+					value: "failed",
+					displayed: true,
+			)
+		}
+	} else {
+		def cmd = zwave.parse(description, [0x98: 1, 0x20: 1, 0x70: 1])
+		if (cmd) {
+			result = zwaveEvent(cmd)
+		}
+	}
+	log.debug "Parse returned ${result?.inspect()}"
+	return result
+}
+
+def zwaveEvent(securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand([0x20: 1, 0x85: 2, 0x70: 1])
+	// log.debug "encapsulated: $encapsulatedCommand"
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	}
+}
+
+def zwaveEvent(basicv1.BasicReport cmd) {
+	log.debug "rx $cmd"
+	handleDeviceEvent(cmd.value)
+}
+
+def zwaveEvent(notificationv3.NotificationReport cmd) {
+	def result = []
+
+	log.debug "rx $cmd"
+
+	if (cmd.notificationType == notificationv3.NotificationReport.NOTIFICATION_TYPE_BURGLAR) {
+		if (cmd.event == 3) {
+			result << createEvent(name: "tamper", value: "detected")
+		}
+		result = result + handleDeviceEvent(cmd.event)
```

Comment:
this should be in an `else`, shouldn't it?

as-is, `handleDeviceEvent` will report the switch and alarm on after a tamper.

Reply:
Originally I had it only handle the tamper (we only seem to get the notification reports for tamper anyway), but my thinking here is that since tamper sets the alarm off and the only way tamper clears is by going in and turning the alarm off it might be best to also send the alarm and switch events.

So the logic being:

Send alarm command:
alarm: on
switch: on

Send chime command:
chime: on

Receive tamper:
tamper: on
alarm: on
switch: on

Something turns alarm, chime, or tamper off:
everything: off

The chime is interesting, too (and I think I described it in a comment), because we never receive any "on" states, so I kind of have to fudge that.

Reply:
See my comment below on Sensor Binary Report. I've updated this.